pkg/monitor/monitor/monitor_manager.go

```diff
@@ -82,7 +86,26 @@ func NewMonitorManager(
 func (mm *MonitorManager) SyncMonitor(monitor *v1alpha1.TidbMonitor) error {
 
 	if monitor.DeletionTimestamp != nil {
-		return nil
+		return mm.patchTidbClusterStatus(nil, monitor)
+	}
+	if monitor.Spec.Clusters == nil || len(monitor.Spec.Clusters) < 1 {
+		err := fmt.Errorf("tm[%s/%s] haven't point the target tidbcluster", monitor.Namespace, monitor.Name)
+		return err
+	}
+	tcRef := monitor.Spec.Clusters[0]
+	if len(tcRef.Namespace) < 1 {
+		tcRef.Namespace = monitor.Namespace
+	}
+	tc, err := mm.cli.PingcapV1alpha1().TidbClusters(tcRef.Namespace).Get(tcRef.Name, metav1.GetOptions{})
```

Comment:
Is it necessary to call kube-apiserver directly here?
What's the drawback of the lister?
We should avoid call kube-apiserver directly especially when too many clusters managed by tidb-operator.

Reply:
I guess that's ok as we only sync tidbmonitor 30s once in default.

Reply:
What if we have 30 tidbmonitors in the cluster, btw, we also have other codes call kube-apiserver directly, which will cause pressure to the kube-apiserver, so I think if not required, we can use the lister, and for the two tidbmonitor to monitor the same tidbcluster case, one of the request to update will fail anyway, right?